src/include/io.h

```diff
@@ -0,0 +1,16 @@
+#ifndef _IO_H
+#define _IO_H 1
+
+static __inline void outb(unsigned char value, unsigned short int port)
+{
+   __asm__ __volatile__ ("outb %w1,%b0": :"a" (value), "Nd" (port));
```

Comment:
Wondering what these cosntraints "a" and "Nd" mean. Looking at https://gcc.gnu.org/onlinedocs/gcc/Simple-Constraints.html#Simple-Constraints , seems like 'd', 'a', and 'f' are supported by the 68k. Are these right for amd64?

Reply:
Yeah, the Simple Constraints page looks like it keeps specifying things about 68k. If you check out the Machine-specific constraints you'll see they're overloaded for x86/64 https://gcc.gnu.org/onlinedocs/gcc/Machine-Constraints.html .  For x86, `a, b, c, d, si, di, bp, sp` refer to the related GPR `a` = `eax`, `b` = `ebx`, etc.

Reply:
Also, forgot to add, `N` for x86 constraints is an "Unsigned 8-bit integer constant (for `in` and `out` instructions)."